packages/assets-controllers/src/utils/formatters.ts

```diff
@@ -51,6 +51,30 @@ function getCachedNumberFormat(
   return format;
 }
 
+/**
+ * Format a number with optional Intl overrides.
+ *
+ * @param config - Configuration object with locale.
+ * @param config.locale - Locale string.
+ * @param value - Numeric value to format.
+ * @param options - Optional Intl.NumberFormat overrides.
+ * @returns Formatted number string.
+ */
+function formatNumber(
+  config: { locale: string },
+  value: number | bigint | `${number}`,
+  options: Intl.NumberFormatOptions = {},
+) {
+  if (!Number.isFinite(Number(value))) {
+    return '';
+  }
+
+  const numberFormat = getCachedNumberFormat(config.locale, options);
+
+  // @ts-expect-error Remove this comment once TypeScript is updated to 5.5+
+  return numberFormat.format(value);
```

Comment:
Hmm this comment is not fixed on typescript 5.5+

[Playground Link](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBGCuBbARgUwE4DETsQQ1gF45UB3GASTCgBsA6AOSTSx3ygAoBKAKB4DN4YYFACW4GPzYEmKDBx4wYoMP1EBzAFwwA3jBohgeGqm3R0osOpgBfADSKYAN2PxTcZhhgAfGMg2WsL4ABgAkOghy6DbBDkogAA5i4BDaVLSMnqy4BADySeKQMMQ69jxcuo6i-DAcAISyLHSiEJiWolCoHI3yLjRuXIOVSkroqFDw6GAwAOQzANyONnxKKtAwAB7FswuOYxNTHlHYOVB0UqccfQOLNkA)

Reply:
Lemme do a short look if we can remove this comment.

Reply:
@n3ps you sure? You may need to restart the TS LSP once you change versions.
I'm testing with 5.9.2 locally and still errors.
<img width="913" height="571" alt="Screenshot 2025-09-19 at 15 25 45" src="https://github.com/user-attachments/assets/3efd5f0b-1a2d-4fa6-a849-1b87145569ca" />